prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusTimeSeries.java

```diff
@@ -0,0 +1,186 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.connector.prometheus.sink;
+
+import org.apache.flink.annotation.Public;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Pojo used as sink input, containing a single TimeSeries: a list of Labels and a list of Samples.
+ *
+ * <p>metricName is mapped in Prometheus to the value of the mandatory label named '__name__'
+ * labels. The other labels, as key/value, are appended after the '__name__' label.
+ */
+@Public
+public class PrometheusTimeSeries implements Serializable {
+    /** A single Label. */
+    public static class Label implements Serializable {
+        private final String name;
+        private final String value;
+
+        public Label(String name, String value) {
```

Comment:
Can we add some `null`/`empty` validations for these data classes? (+ unit  tests too!)

Reply:
There other other validations that are more critical, like Labels must be in order by name and samples in order by timestamp, or Prometheus will reject them.
We kept these validations out of the implementation on purpose, because this object will be called a gazillion times, and this may impact performance.
In the docs, we explicitly say this is a responsibility of the application, like sending the time series in time order.

Either we decide to add validation to this class, paying the performance fee, or we leave this class unvalidated

Reply:
I added further explanations in the README about the behavior and the responsibility of sending well-formed data.
In any case, with the default error-handling behavior, if the application sends malformed data, the sink throws an exception when Prometheus rejects the write.